module-4/metascan/README.md

```diff
@@ -0,0 +1,94 @@
+# PDF Metadata Extractor
+
+This Python script downloads PDF files from a specified website and extracts metadata such as title, author, creation date, etc., saving them into a CSV file.
+
+## Features
+
+- **Web Scraping**: Extracts all PDF links from a given website using `BeautifulSoup` and `requests` modules.
+- **PDF Download**: Downloads each PDF file found to a local directory (`pdfs/`).
+- **Metadata Extraction**: Uses the `PyPDF2` library to extract metadata (title, author, creation date, etc.) from each downloaded PDF file.
+- **CSV Export**: Saves extracted metadata into a CSV file with semicolon (`;`) as a delimiter for better readability.
+- **Error Handling**: Provides basic error handling for network requests and PDF metadata extraction failures.
+- **Command Line Interface**: Supports command-line arguments for specifying the website URL (`-u`) and output CSV file name (`-n`).
+
+## Function Descriptions
+
+### `extract_pdf_links(url)`
+
+This function takes a URL as an input and extracts all PDF links from the webpage. It uses the `requests` module to fetch the page content and `BeautifulSoup` to parse the HTML.
+
+- **Parameters**:
+  - `url` (str): The URL of the webpage to scan for PDF files.
+- **Returns**:
+  - `pdf_links` (list): A list of URLs to PDF files found on the webpage.
+
+### `download_pdf(url, save_path)`
+
+This function downloads a PDF file from the provided URL and saves it to the specified path.
+
+- **Parameters**:
+  - `url` (str): The URL of the PDF file to download.
+  - `save_path` (str): The local file path to save the downloaded PDF.
+- **Outputs**:
+  - The function prints a confirmation message indicating the download is complete.
+
+### `extract_pdf_metadata(file_path)`
+
+This function extracts metadata from a PDF file. It reads the PDF file using `PyPDF2` and retrieves various metadata attributes.
+
+- **Parameters**:
+  - `file_path` (str): The local file path of the PDF file to extract metadata from.
+- **Returns**:
+  - `metadata` (dict): A dictionary containing metadata fields such as title, author, creation date, etc.
+
+### `main()`
+
+This is the main function that orchestrates the entire process. It parses command-line arguments, extracts PDF links from the given URL, downloads each PDF, extracts metadata, and saves the results into a CSV file.
+
+- **Command-Line Arguments**:
+  - `-u` / `--url` (str): The URL of the website to scan for PDF files.
+  - `-n` / `--name` (str): The name and path of the output CSV file.
+- **Outputs**:
+  - The function prints messages to indicate progress, such as downloaded files and saved metadata.
+
+## Usage
+
+1. **Setup Environment**:
+    - Ensure Python 3.x and necessary libraries (`requests`, `beautifulsoup4`, `PyPDF2`) are installed.
```

Comment:
How would I do that? 

make this instructional:
--> Ensure Python is installed in your computer with the command `python --version`
--> install requirements via pip with the command: ...
    - requires pip to be installed, can be installed with: `...`